### PR TITLE
Add Unit Usage support to Number Skeletons.

### PR DESCRIPTION
**Pull request by hugovdm**
##### Checklist

- [ ] Issue filed: https://unicode-org.atlassian.net/browse/ICU-_____
- [ ] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/i18n/unitsdata.cpp [hunk 1]
```diff
@@ -329,6 +329,10 @@ int32_t getPreferenceMetadataIndex(const MaybeStackVector<UnitPreferenceMetadata
         } else if (uprv_strcmp(desired.usage.data(), "default") != 0) {
             desired.usage.truncate(0).append("default", status);
         } else {
+            // TODO(units): we fail with U_MISSING_RESOURCE_ERROR for an invalid
+            // usage. For an unrecognised measure-unit, skeleton parsing would
+            // result in U_NUMBER_SKELETON_SYNTAX_ERROR instead. Do we want to
+            // be smarter about recognised usages? (Build a trie? :-)
```
Comment (sffc) [inline icu4c/source/i18n/unitsdata.cpp:335]: This is worth some discussion.  Can you open a follow-up issue and reference it here?

`// TODO(icu-units/icu#123): blah blah`

File: icu4c/source/test/intltest/numbertest_api.cpp [hunk 2]
```diff
@@ -687,73 +687,65 @@ void NumberFormatterApiTest::unitUsage() {
...
-//                               u"877 km",
-//                               u"88 km",
-//                               u"8,8 km",
-//                               u"877 m",
-//                               u"88 m",
-//                               u"8,8 m",
-//                               u"0 m");
+    assertFormatDescendingBig(
+            u"unitUsage() en-ZA road",
+            u"measure-unit/length-meter usage/road", // Alternatives: usage/length-road, unit-usage/road? FIXME
```
Comment (sffc) [inline icu4c/source/test/intltest/numbertest_api.cpp:692]: `usage/road` is fine.  It matches the command-line argument.

File: icu4c/source/test/intltest/numbertest_api.cpp [hunk 3]
```diff
@@ -687,73 +687,65 @@ void NumberFormatterApiTest::unitUsage() {
...
-//                               u"545 mi",
-//                               u"54 mi",
-//                               u"5.4 mi",
-//                               u"0.54 mi",
-//                               u"96 yd",
-//                               u"9.6 yd",
-//                               u"0 yd");
+    assertFormatDescendingBig(
+            u"unitUsage() en-GB road",
+            u"measure-unit/length-meter usage/road", // Alternatives: usage/length-road, unit-usage/road? FIXME
```
Comment (sffc) [inline icu4c/source/test/intltest/numbertest_api.cpp:714]: Delete the comment

File: icu4c/source/i18n/number_skeletons.cpp [hunk 4]
```diff
@@ -993,6 +1006,8 @@ void blueprint_helpers::parseMeasureUnitOption(const StringSegment& segment, Mac
     static constexpr int32_t CAPACITY = 30;
     MeasureUnit units[CAPACITY];
     UErrorCode localStatus = U_ZERO_ERROR;
+    // TODO(units): revisit this: deals with hard-coded set of measure units?
+    // Can it handle complex units correctly?
```
[the remark this answers is not in the document]
Comment (hugovdm) [reply icu4c/source/i18n/number_skeletons.cpp]: Done.

Issue: https://github.com/icu-units/icu/issues/35

File: icu4c/source/i18n/number_skeletons.cpp [hunk 5]
```diff
@@ -1545,6 +1571,15 @@ bool GeneratorHelpers::perUnit(const MacroProps& macros, UnicodeString& sb, UErr
     }
 }
 
+bool GeneratorHelpers::usage(const MacroProps& macros, UnicodeString& sb, UErrorCode& status) {
+    if (macros.usage.fLength > 0) {
+        sb.append(u"usage/", -1);
+        sb.append(UnicodeString(macros.usage.fUsage));
```
[the remark this answers is not in the document]
Comment (hugovdm) [reply icu4c/source/i18n/number_skeletons.cpp]: Done.

File: icu4c/source/i18n/number_skeletons.h [hunk 6]
```diff
@@ -46,7 +48,8 @@ enum ParseState {
     STATE_INCREMENT_PRECISION,
     STATE_MEASURE_UNIT,
     STATE_PER_MEASURE_UNIT,
-    STATE_IDENTIFIER_UNIT,
+    STATE_IDENTIFIER_UNIT, // A MeasureUnit Identifier. TODO(units): Unused?
```
[the remark this answers is not in the document]
Comment (hugovdm) [reply icu4c/source/i18n/number_skeletons.h]: Thanks - deleted again.

(I can't think how I missed some of the key references this on my first pass. :) Rushing, I suppose.)

File: icu4c/source/i18n/unitsdata.cpp (hunk 1 above)
Comment (hugovdm) [reply icu4c/source/i18n/unitsdata.cpp]: Done.

File: icu4c/source/test/intltest/numbertest_api.cpp (hunk 2 above)
Comment (hugovdm) [reply icu4c/source/test/intltest/numbertest_api.cpp]: Done.

File: icu4c/source/test/intltest/numbertest_api.cpp (hunk 3 above)
Comment (hugovdm) [reply icu4c/source/test/intltest/numbertest_api.cpp]: Done.

**Pull request merged**